src/main/java/dinkplugin/util/WorldTypeTracker.java

```diff
@@ -0,0 +1,35 @@
+package dinkplugin.util;
+
+import dinkplugin.domain.SeasonalPolicy;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.WorldType;
+
+import javax.inject.Singleton;
+import java.util.Set;
+
+@Slf4j
+@Singleton
+public class WorldTypeTracker extends BooleanStateTracker {
+
+    @Override
+    protected void populateState() {
+        Set<WorldType> world = client.getWorldType();
+        if (config.seasonalPolicy() == SeasonalPolicy.REJECT && world.contains(WorldType.SEASONAL)) {
+            this.state = false;
+        } else {
+            this.state = !WorldUtils.isIgnoredWorld(world);
+        }
```

Comment:
imo this is more readable, so I'd prefer this. not a hill I'll die on though

```suggestion
        } else if (WorldUtils.isIgnoredWorld(world) {
            this.state = false;
        } else {
            this.state = true;
        }
```